OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -622,10 +622,15 @@ bool AircraftCanResupplyAt(Actor a, bool allowedToForceEnter = false)
 
 		public (CPos Cell, SubCell SubCell)[] OccupiedCells()
 		{
-			if (self.World.Map.DistanceAboveTerrain(CenterPosition).Length >= Info.MinAirborneAltitude)
-				return landingCells.Select(c => (c, SubCell.FullCell)).ToArray();
+			return OccupiedCells(CenterPosition);
+		}
+
+		(CPos Cell, SubCell SubCell)[] OccupiedCells(WPos pos)
+		{
+			if (self.World.Map.DistanceAboveTerrain(pos).Length >= Info.MinAirborneAltitude)
+				return landingCells;
 
-			return new[] { (TopLeft, SubCell.FullCell) };
+			return new[] { (self.World.Map.CellContaining(pos), SubCell.FullCell) };
```

Comment:
This check should really just be 
```C#
return landingCells
```

We only occupy space when we have landing cells set. If we don't then we shouldn't be occupying space in the first place.

This solution also addresses the issue of needing to check for changes in SetPosition

Reply:
Actually we should still have checks in SetPosition

If landingCells has value and we are below equal to Info.MinAirborneAltitude and our Position doesn't match to the occupied cells we should update landedCells

Reply:
Why do the landingCells differ from our Position in the first place, shouldn't things that change our landing spot also update the landingCells alongside the Position?

Reply:
planes want to reserve their landing position, but as don't land vertically they can't reserve on their current position

Reply:
> This check should really just be

Also see https://github.com/OpenRA/OpenRA/pull/20212#discussion_r951822741

Reply:
Agree with the above, making this change as suggested would break pre-placed actors, so will leave as-is.

Reply:
My suggested change would not break that, as we would only check if `landingCells` match TopLeft  if we have landed. Basically the same thing we are doing just with much cleaner and better readable code